lib/client/client.go

```diff
@@ -178,6 +207,24 @@ func (proxy *ProxyClient) ReissueUserCerts(ctx context.Context, params ReissuePa
 	return trace.Wrap(err)
 }
 
+// RootClusterName returns name of the current cluster
+func (proxy *ProxyClient) RootClusterName() (string, error) {
+	localAgent := proxy.teleportClient.LocalAgent()
```

Comment:
if I remember correctly, a client can run with a nil agent (e.g. when using an identity file)
it has caused a few bugs in the past, with nil pointer dereferences.

Reply:
it's also used in `ConnectToCluster` and `ReissueCerts` and I'm not quite sure what to use instead if this is not available. I think you are right that `tsh --i clusters ls` will fail with or without this addition. I have created a separate ticket to fix this and other instances. https://github.com/gravitational/teleport/issues/4669